python/pyspark/pandas/sql_formatter.py

```diff
@@ -265,7 +266,14 @@ def _convert_value(self, val: Any, name: str) -> Optional[str]:
             val._to_spark().createOrReplaceTempView(df_name)
             return df_name
         elif isinstance(val, str):
-            return lit(val)._jc.expr().sql()  # for escaped characters.
+            if is_remote():
+                special_characters = ["\\", "'", '"', "\0", "\b", "\n", "\r", "\t"]
+                escaped_val = val
+                for char in special_characters:
+                    escaped_val = escaped_val.replace(char, "\\" + char)
+                return f"'{escaped_val}'"
```

Comment:
This is too hacky. Is this exactly matched with JVM implementaton?

Reply:
I cannot find a workaround based on DF apis, not sure whether a new proto message is needed

Reply:
Yeah, actually I just refer the implementation from JVM to handle the `str` as below:

```scala
    case (v: UTF8String, StringType) =>
      // Escapes all backslashes and single quotes.
      "'" + v.toString.replace("\\", "\\\\").replace("'", "\\'") + "'"
```

Seems like they only care about the single quotation mark, but I additionally add some more possible special characters into `special_characters`.

Reply:
Okay, can you leave a comment here that this implementation matches with JVM side?